src/core/public/http/fetch.ts

```diff
@@ -133,7 +133,11 @@ export class Fetch {
     try {
       response = await window.fetch(request);
     } catch (err) {
-      throw new HttpFetchError(err.message, request);
+      if (err.name === 'AbortError') {
```

Comment:
let's merge as-is. I don't see any tests for abortable requests. @elastic/kibana-platform can add them in https://github.com/elastic/kibana/issues/51406 
This test fails with timeout locally: 
```js
  it('should accept abort request signal', async () => {
      expect.assertions(1);

      fetchMock.get('*', () => new Promise(resolve => setTimeout(() => resolve('a'), 50000)));

      const abortController = new window.AbortController();
      const { signal } = abortController;
      setTimeout(() => abortController.abort(), 100);

      try {
        await fetchInstance.fetch('/my/path', { signal });
      } catch (e) {
        expect(e.name).toBe('AbortError');
      }
  });
```